### PR TITLE
Add header

### PR DESCRIPTION
**Pull request by Adam-Oueidat**

### CODE REVIEW

File: apps/nextjs/src/app/_components/header.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import Link from "next/link";
+
+import { auth, signIn, signOut } from "@acme/auth";
+import { Button } from "@acme/ui/button";
+
+const sharedClasses = {
```
Comment (GNRSN) [inline apps/nextjs/src/app/_components/header.tsx:6]: FYI: I think this is a tailwind antipattern. 
I'm unsure if lsp and formatting will handle this correctly

**Comment by GNRSN** [comment]
Weirdly enough, auth isn't working in the preview build. I've validated that it works in the preview of other PRs though, e.g. [#3 ](https://shared-expensesnext-js-git-pre-commit-fo-147761-gnrsns-projects.vercel.app/)

File: apps/nextjs/src/app/_components/header.tsx (hunk 1 above)
Comment (Adam-Oueidat) [reply apps/nextjs/src/app/_components/header.tsx:6]: The shared classes? I agree 100%, I think this was me experimenting with Co-pilot and wanted it to improve my code but it rather made an anti-pattern, can be changed

Comment (Adam-Oueidat) [reply apps/nextjs/src/app/_components/header.tsx]: Fixed, close if you are satisfied

**Pull request merged**